src/Framework/Telemetry/LoggingConfigurationTelemetry.cs

```diff
@@ -97,54 +100,65 @@ internal class LoggingConfigurationTelemetry : TelemetryBase
-        {
-            Properties["TerminalLoggerUserIntentSource"] = TerminalLoggerUserIntentSource;
-        }
-
-        if (TerminalLoggerDefault != null)
-        {
-            Properties["TerminalLoggerDefault"] = TerminalLoggerDefault;
-        }
-
-        if (TerminalLoggerDefaultSource != null)
-        {
-            Properties["TerminalLoggerDefaultSource"] = TerminalLoggerDefaultSource;
-        }
-
-        Properties["ConsoleLogger"] = ConsoleLogger.ToString(CultureInfo.InvariantCulture);
-        if (ConsoleLoggerType != null)
-        {
-            Properties["ConsoleLoggerType"] = ConsoleLoggerType;
-        }
-
-        if (ConsoleLoggerVerbosity != null)
-        {
-            Properties["ConsoleLoggerVerbosity"] = ConsoleLoggerVerbosity;
-        }
-
-        Properties["FileLogger"] = FileLogger.ToString(CultureInfo.InvariantCulture);
-        if (FileLoggerType != null)
-        {
-            Properties["FileLoggerType"] = FileLoggerType;
-            Properties["FileLoggersCount"] = FileLoggersCount.ToString(CultureInfo.InvariantCulture);
+            return new Dictionary<string, string>(Properties);
         }
+    }
 
-        if (FileLoggerVerbosity != null)
+    public override void UpdateEventProperties()
+    {
+        lock (_propertiesLock)
         {
-            Properties["FileLoggerVerbosity"] = FileLoggerVerbosity;
+            Properties["TerminalLogger"] = TerminalLogger.ToString(CultureInfo.InvariantCulture);
+
+            if (TerminalLoggerUserIntent != null)
+            {
+                Properties["TerminalLoggerUserIntent"] = TerminalLoggerUserIntent;
+            }
+
+            if (TerminalLoggerUserIntentSource != null)
+            {
+                Properties["TerminalLoggerUserIntentSource"] = TerminalLoggerUserIntentSource;
+            }
+
+            if (TerminalLoggerDefault != null)
+            {
+                Properties["TerminalLoggerDefault"] = TerminalLoggerDefault;
+            }
+
+            if (TerminalLoggerDefaultSource != null)
+            {
+                Properties["TerminalLoggerDefaultSource"] = TerminalLoggerDefaultSource;
+            }
+
+            Properties["ConsoleLogger"] = ConsoleLogger.ToString(CultureInfo.InvariantCulture);
+            if (ConsoleLoggerType != null)
+            {
+                Properties["ConsoleLoggerType"] = ConsoleLoggerType;
+            }
+
+            if (ConsoleLoggerVerbosity != null)
+            {
+                Properties["ConsoleLoggerVerbosity"] = ConsoleLoggerVerbosity;
+            }
+
+            Properties["FileLogger"] = FileLogger.ToString(CultureInfo.InvariantCulture);
+            if (FileLoggerType != null)
+            {
+                Properties["FileLoggerType"] = FileLoggerType;
+                Properties["FileLoggersCount"] = FileLoggersCount.ToString(CultureInfo.InvariantCulture);
+            }
+
+            if (FileLoggerVerbosity != null)
+            {
```

Comment:
nit: I would recommend to create some generic method like :
```
 void SetPropertyIfNotNull(string propertyName, object value)
{
    if (value != null)
    {
        Properties[propertyName] = value;
    }
}
```

Reply:
Thank you for the looking into the PR! 
Do you mind if this kind of refactoring/improvements would be made in separate PRs?

Reply:
As you wish :)
It's just a suggestion, feel free to ignore

Reply:
Thank you!  just wanted to keep the changes as small as possible :) Will definetly address the suggestion!